packages/edit-site/src/hooks/push-changes-to-global-styles/index.js

```diff
@@ -96,27 +98,33 @@ const STYLE_PATH_TO_PRESET_BLOCK_ATTRIBUTE = {
 function useChangesToPush( name, attributes ) {
 	const supports = useSupportedStyles( name );
 
-	return useMemo(
-		() =>
-			supports.flatMap( ( key ) => {
-				if ( ! STYLE_PROPERTY[ key ] ) {
-					return [];
-				}
-				const { value: path } = STYLE_PROPERTY[ key ];
-				const presetAttributeKey = path.join( '.' );
-				const presetAttributeValue =
-					attributes[
-						STYLE_PATH_TO_PRESET_BLOCK_ATTRIBUTE[
-							presetAttributeKey
-						]
-					];
-				const value = presetAttributeValue
-					? `var:preset|${ STYLE_PATH_TO_CSS_VAR_INFIX[ presetAttributeKey ] }|${ presetAttributeValue }`
-					: get( attributes.style, path );
-				return value ? [ { path, value } ] : [];
-			} ),
-		[ supports, name, attributes ]
-	);
+	return useMemo( () => {
+		const changes = supports.flatMap( ( key ) => {
+			if ( ! STYLE_PROPERTY[ key ] ) {
+				return [];
+			}
+			const { value: path } = STYLE_PROPERTY[ key ];
+			const presetAttributeKey = path.join( '.' );
+			const presetAttributeValue =
+				attributes[
+					STYLE_PATH_TO_PRESET_BLOCK_ATTRIBUTE[ presetAttributeKey ]
+				];
+			const value = presetAttributeValue
+				? `var:preset|${ STYLE_PATH_TO_CSS_VAR_INFIX[ presetAttributeKey ] }|${ presetAttributeValue }`
+				: get( attributes.style, path );
+			return value ? [ { path, value } ] : [];
+		} );
+
+		// To ensure display of a visible border, global styles require a
+		// default border style if a border color or width is present.
+		const { color, style, width } = attributes.style?.border || {};
+
+		if ( ( color || width ) && ! style ) {
+			changes.push( { path: [ 'border', 'style' ], value: 'solid' } );
+		}
+
+		return changes;
+	}, [ supports, attributes ] );
```

Comment:
Unused dependency removed to satisfy linter. I couldn't see any regressions but worth flagging in case I missed something.